lib/pages/chat/events/message_download_content.dart

```diff
@@ -92,17 +94,37 @@ class _MessageDownloadContentState extends State<MessageDownloadContent>
             filePath: success.filePath,
           );
         } else if (success is DownloadMatrixFileSuccessState) {
-          downloadFileStateNotifier.value = FileWebDownloadedPresentationState(
-            matrixFile: success.matrixFile,
-          );
+          _handleDownloadMatrixFileSuccessState(success);
         }
       },
     );
   }
 
+  void _handleDownloadMatrixFileSuccessState(
+    DownloadMatrixFileSuccessState success,
+  ) {
+    streamSubscription?.cancel();
+    if (mounted) {
+      downloadFileStateNotifier.value = FileWebDownloadedPresentationState(
+        matrixFile: success.matrixFile,
+      );
+      handlePreviewWeb(event: widget.event, context: context);
+      return;
+    }
+
+    if (TwakeApp.routerKey.currentContext != null) {
+      handlePreviewWeb(
+        event: widget.event,
+        context: TwakeApp.routerKey.currentContext!,
+      );
+    }
+  }
+
   @override
   void dispose() {
-    streamSubscription?.cancel();
+    if (!PlatformInfos.isWeb) {
```

Comment:
why not cancel the subscription on web ?

Reply:
we cancel after the file is downloaded, because we want to download the file even if user change to another chat